server/src/main/java/io/crate/metadata/cluster/AlterTableClusterStateExecutor.java

```diff
@@ -111,6 +111,7 @@ public AlterTableClusterStateExecutor(MetadataMappingService metadataMappingServ
     protected ClusterState execute(ClusterState currentState, AlterTableRequest request) throws Exception {
         if (request.isPartitioned()) {
             if (request.partitionIndexName() != null) {
+                // ADD COLUMN never goes through this path
```

Comment:
why not?

Reply:
I think if you follow the places that `AlterTableRequest` is created, you can see that that `if` will always fail for ADD COLUMN. Anyways, I will remove the comment.

Reply:
If possible code should guard against being used in a way it cannot handle correctly. The analyzer/planner code might change and then the comment here would get outdated and likely cause incorrect behavior.

I opened https://github.com/crate/crate/pull/12821